cmd/influx/v1_dbrp.go

```diff
@@ -0,0 +1,386 @@
+package main
+
+import (
+	"context"
+	"io"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/cmd/influx/internal"
+	"github.com/influxdata/influxdb/v2/dbrp"
+	"github.com/spf13/cobra"
+)
+
+func cmdV1DBRP(f *globalFlags, opt genericCLIOpts) *cobra.Command {
+	cmd := opt.newCmd("dbrp", nil, false)
+	cmd.Aliases = []string{"dbrp"}
+	cmd.Short = "Database retention policy mappings for v1 APIs"
+	cmd.Run = seeHelp
+
+	cmd.AddCommand(
+		v1DBRPCreateCmd(f, opt),
+		v1DBRPFindCmd(f, opt),
+		v1DBRPDeleteCmd(f, opt),
+		v1DBRPUpdateCmd(f, opt),
+	)
+
+	return cmd
+}
+
+var v1DBRPCRUDFlags struct {
+	id          string
+	json        bool
+	hideHeaders bool
+}
+
+var v1DBRPFindFlags struct {
+	BucketID string       // Specifies the bucket ID to filter on
+	DB       string       // Specifies the database to filter on
+	Default  bool         // Specifies filtering on default
+	ID       string       // Specifies the mapping ID to filter on
+	Org      organization // required  // Specifies the organization ID to filter on
+	RP       string       // Specifies the retention policy to filter on
+}
+
+func v1DBRPFindCmd(f *globalFlags, opt genericCLIOpts) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List DBRP",
+		RunE:  checkSetupRunEMiddleware(&flags)(v1DBRPFindF),
+	}
+
+	f.registerFlags(opt.viper, cmd)
+	registerPrintOptions(opt.viper, cmd, &v1DBRPCRUDFlags.hideHeaders, &v1DBRPCRUDFlags.json)
+	v1DBRPFindFlags.Org.register(opt.viper, cmd, false)
+	cmd.Flags().StringVarP(&v1DBRPCreateFlags.BucketID, "bucket-id", "b", "", "the bucket ID to filter on")
+	cmd.Flags().StringVarP(&v1DBRPCreateFlags.DB, "db", "d", "", "the v1 database to map from")
+	cmd.Flags().BoolVar(&v1DBRPCreateFlags.Default, "default", false, "Specify if this mapping represents the default retention policy for the database specificed")
+	cmd.Flags().StringVarP(&v1DBRPCreateFlags.RP, "rp", "r", "", "InfluxDB v1 retention policy")
+
+	return cmd
+}
+
+func v1DBRPFindF(cmd *cobra.Command, args []string) error {
+	if err := v1DBRPFindFlags.Org.validOrgFlags(&flags); err != nil {
+		return err
+	}
+	orgSvc, err := newOrganizationService()
+	if err != nil {
+		return err
+	}
+
+	orgID, err := v1DBRPCreateFlags.Org.getID(orgSvc)
+	if err != nil {
+		return err
+	}
+
+	s, err := newV1DBRPService()
+	if err != nil {
+		return err
+	}
+	filter := influxdb.DBRPMappingFilterV2{OrgID: &orgID}
+
+	if err := filter.OrgID.DecodeFromString(orgID.String()); err != nil {
```

Comment:
Could you add a comment explaining why this is needed? Without knowing the code, I'm surprised that the ID returned by `v1DBRPCreateFlags.Org.getId` isn't usable as-is.

Reply:
Alt: I see in the delete command that the `Org.getId` result is used as-is, could we delete this statement entirely?